File: Leet Code/JavaScript/problem_258.js
const addDigits = function(num) {
    if (num < 10) {
        return num;
    };

    let numSum = num;
    while (numSum >= 10) {
        let sum = digitSum(numSum);
        numSum = sum;
    };
    
    return numSum;
    
};

const digitSum = function(number) {
    let sum = 0;
    while (number > 0) {
        let last = number % 10;
        sum += last;
        number = Math.floor(number / 10);
    };
    return sum;
};

const result = addDigits(38);
console.log(result);

File: Leet Code/Java/Problem258.java
import java.util.Scanner;

public class Problem258 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter number:");
        int num = in.nextInt();
        int result = addDigits(num);
        System.out.println(result);
    }

    static int addDigits(int num) {
        if (num < 10) {
            return num;
        }
        int numSum = num;
        while (numSum >= 10) {
            int newNum = digitSum(numSum);
            numSum = newNum;
        }
        return numSum;
    }

    static int digitSum(int num) {
        int sum = 0;
        while (num > 0) {
            int last = num % 10;
            sum += last;
            num /= 10;
        }
        return sum;
    }
}
